File: scripts/convert-fonts.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Convert TTF/OTF fonts to WOFF2 format
Requires: pip install fonttools[woff]
"""

import os
import sys
from pathlib import Path

# Fix encoding for Windows console
if sys.platform == 'win32':
    import codecs
    sys.stdout = codecs.getwriter('utf-8')(sys.stdout.buffer, 'strict')
    sys.stderr = codecs.getwriter('utf-8')(sys.stderr.buffer, 'strict')

try:
    from fontTools.ttLib import TTFont
except ImportError:
    print("[!] fonttools not installed. Installing...")
    os.system(f"{sys.executable} -m pip install fonttools[woff] --quiet")
    from fontTools.ttLib import TTFont

ROOT = Path(__file__).parent.parent
FONTS_DIR = ROOT / "frontend" / "public" / "fonts"

FONTS_TO_CONVERT = [
    "salena-regular.ttf",
    "salena-medium.ttf",
    "salena-semibold.ttf",
    "salena-bold.ttf",
    "DavigoDemoRegular-V4Bg0.ttf",
]

def convert_to_woff2(input_path, output_path):
    """Convert TTF/OTF to WOFF2"""
    try:
        font = TTFont(input_path)
        font.flavor = 'woff2'
        font.save(output_path)
        return True
    except Exception as e:
        print(f"   Error: {e}")
        return False

def main():
    print("Converting fonts to WOFF2...\n")
    
    converted = 0
    failed = 0
    
    for font_file in FONTS_TO_CONVERT:
        input_path = FONTS_DIR / font_file
        
        if not input_path.exists():
            # Try .otf extension
            input_path = input_path.with_suffix('.otf')
            if not input_path.exists():
                print(f"[!] Not found: {font_file}")
                failed += 1
                continue
        
        output_name = font_file.replace('.ttf', '.woff2').replace('.otf', '.woff2')
        output_path = FONTS_DIR / output_name
        
        print(f"Converting: {font_file} -> {output_name}...", end=" ")
        
        if convert_to_woff2(input_path, output_path):
            # Get file sizes
            input_size = input_path.stat().st_size / 1024
            output_size = output_path.stat().st_size / 1024
            reduction = ((input_size - output_size) / input_size) * 100
            print(f"OK ({input_size:.1f}KB -> {output_size:.1f}KB, -{reduction:.1f}%)")
            converted += 1
        else:
            print("FAILED")
            failed += 1
    
    print(f"\n[+] Conversion complete!")
    print(f"   Converted: {converted}")
    if failed > 0:
        print(f"   Failed: {failed}")

if __name__ == "__main__":
    main()

File: scripts/convert-fonts.js
const fs = require('fs-extra');
const path = require('path');

const ROOT = path.resolve(__dirname, '..');
const SOURCE_DIR = path.join(ROOT, 'frontend', 'src', 'components');
const TARGET_DIR = path.join(ROOT, 'frontend', 'public', 'fonts');

// Fonts to convert (only the ones we actually use)
const FONTS_TO_CONVERT = [
  // Salena - only used weights (removed italic and light variants)
  { name: 'salena-regular', source: 'salena-regular.ttf', weight: 400 },
  { name: 'salena-medium', source: 'salena-medium.ttf', weight: 500 },
  { name: 'salena-semibold', source: 'salena-semibold.ttf', weight: 600 },
  { name: 'salena-bold', source: 'salena-bold.ttf', weight: 700 },
  // Davigo - only regular
  { name: 'davigo-regular', source: 'DavigoDemoRegular-V4Bg0.ttf', weight: 400, dir: 'davigo-font' },
];

async function prepareFonts() {
  console.log('Preparing fonts for conversion...\n');
  
  // Ensure target directory exists
  await fs.ensureDir(TARGET_DIR);

  const fontsToConvert = [];

  for (const font of FONTS_TO_CONVERT) {
    const sourceDir = font.dir ? path.join(SOURCE_DIR, font.dir) : path.join(SOURCE_DIR, 'salena-font');
    const sourceFile = path.join(sourceDir, font.source);
    
    // Try TTF first, then OTF
    let actualSource = sourceFile;
    if (!(await fs.pathExists(actualSource)) && font.source.endsWith('.ttf')) {
      actualSource = sourceFile.replace('.ttf', '.otf');
    }

    if (!(await fs.pathExists(actualSource))) {
      console.warn(`⚠️  Source file not found: ${actualSource}`);
      continue;
    }

    // Copy TTF/OTF to fonts directory for conversion
    const targetTtf = path.join(TARGET_DIR, font.source);
    await fs.copy(actualSource, targetTtf);
    fontsToConvert.push({ name: font.name, file: font.source, path: targetTtf });
    console.log(`✅ Prepared: ${font.source}`);
  }

  console.log(`\n✨ ${fontsToConvert.length} fonts prepared in: ${TARGET_DIR}`);
  console.log('\n📝 Next steps:');
  console.log('   1. Convert TTF/OTF files to WOFF2 using one of these methods:');
  console.log('      - Online: https://cloudconvert.com/ttf-to-woff2');
  console.log('      - Online: https://convertio.co/ttf-woff2/');
  console.log('      - Command line: npx @fontsource/woff2-cli (if available)');
  console.log('   2. Place converted .woff2 files in: frontend/public/fonts/');
  console.log('   3. Ensure filenames match: salena-regular.woff2, salena-medium.woff2, etc.');
  console.log('\n💡 Tip: WOFF2 files are ~30-50% smaller than TTF/OTF!');
}

prepareFonts().catch(console.error);
